src/Controls/src/Core/Handlers/Items/Android/MauiCarouselRecyclerView.cs

```diff
@@ -273,6 +273,11 @@ void CollectionItemsSourceChanged(object sender, System.Collections.Specialized.
 					.Dispatch(() =>
 					{
 
+						if (carouselPosition >= observableItemsSource.Count)
```

Comment:
Debugging through this one it seems like the bug here has to do with multiple collection changed events firing. 

If I break point here

![image](https://github.com/dotnet/maui/assets/5375137/67f001c3-50f3-479f-970e-cfa4ca6790e5)

it fires a bunch every time an item is inserted. 
Either my debugger is being weird or there is something really weird going on 

thoughts?

i think if this collectionchanged event would only fire once this would all work fine as is.

Reply:
I think one of the issues here is with this code

https://github.com/dotnet/maui/blob/aad3c858d8f8c846772e269f2d25b533b2a17200/src/Controls/src/Core/Handlers/Items/Android/MauiCarouselRecyclerView.cs#L131-L136

During startup we recreate the itemsource a bunch of times (we should only create it once)

But when we do create it more than once the old adapters aren't getting unsubscribed from :( which causes a bunch of CollectionChanged events to fire for each single action